src/components/Form/ArrayBrowserField.js

```diff
@@ -24,22 +27,38 @@ function ArrayBrowserField({
       <div className="space-y-2 mt-2">
         <div id="DocumentThumbnailGallery">
           {documentIds?.length > 0 &&
-            documentIds?.map((doc, index) => (
+            documentIds?.map((docId) => (
               <div
-                key={`${doc}_${index}`}
+                key={`${docId}`}
                 className="inline-flex border border-transparent bg-white rounded-lg shadow-md text-sm font-medium p-2 space-x-1 mr-2 mb-2"
               >
-                <DocumentThumbnail.Container docId={doc} />
+                {docType === AUDIO && (
+                  <MediaThumbnail.Audio id={docId} type={docType} />
+                )}
+                {docType === IMAGE && (
+                  <MediaThumbnail.Image
+                    id={docId}
+                    type={docType}
+                    size={SMALL}
+                    imageStyles="object-cover pointer-events-none"
+                  />
+                )}
+                {docType === VIDEO && (
+                  <MediaThumbnail.Video id={docId} type={docType} />
+                )}
+                {/* For all other types of documents. */}
+                {![AUDIO, IMAGE, VIDEO].includes(docType) && (
```

Comment:
Still kept the DocumentThumbnail till we move all documents to new data types and completely do away with the usage of getDocument endpoint.